packages/front-end/components/FactTables/FactMetricModal.tsx

```diff
@@ -405,92 +424,254 @@ function ColumnRefSelector({
+    );
+  }
+  if (
+    type === "quantile" &&
+    quantileSettings.type === "event" &&
+    quantileSettings.ignoreZeros
+  ) {
+    whereParts.push(`-- Ignore zeros in percentile\nvalue > 0`);
+  }
+
+  return whereParts.length > 0
+    ? `\nWHERE\n${indentLines(whereParts.join(" AND\n"))}`
+    : "";
+}
+
+function getPreviewSQL({
+  type,
+  quantileSettings,
+  windowSettings,
+  numerator,
+  denominator,
+  numeratorFactTable,
+  denominatorFactTable,
+}: {
+  type: FactMetricType;
+  quantileSettings: MetricQuantileSettings;
+  windowSettings: MetricWindowSettings;
+  numerator: ColumnRef;
+  denominator: ColumnRef | null;
+  numeratorFactTable: FactTableInterface | null;
+  denominatorFactTable: FactTableInterface | null;
+}): { sql: string; denominatorSQL?: string; experimentSQL: string } {
+  const identifier =
+    "`" + (numeratorFactTable?.userIdTypes?.[0] || "user_id") + "`";
+
+  const identifierComment =
+    (numeratorFactTable?.userIdTypes?.length || 0) > 1
+      ? `\n  -- All of the Fact Table's identifier types are supported`
+      : "";
+
+  const numeratorName = "`" + (numeratorFactTable?.name || "Fact Table") + "`";
+  const denominatorName =
+    "`" + (denominatorFactTable?.name || "Fact Table") + "`";
+
+  const numeratorCol =
+    numerator.column === "$$count"
+      ? "COUNT(*)"
+      : numerator.column === "$$distinctUsers"
+      ? "1"
+      : `SUM(${numerator.column})`;
+
+  const denominatorCol =
+    denominator?.column === "$$count"
+      ? "COUNT(*)"
+      : denominator?.column === "$$distinctUsers"
+      ? "1"
+      : `SUM(${denominator?.column})`;
+
+  const WHERE = getWHERE({
+    factTable: numeratorFactTable,
+    columnRef: numerator,
+    windowSettings,
+    quantileSettings,
+    type,
+  });
+
+  const DENOMINATOR_WHERE = getWHERE({
+    factTable: denominatorFactTable,
+    columnRef: denominator,
+    windowSettings,
+    quantileSettings,
+    type,
+  });
+
+  let HAVING = "";
+  if (type === "quantile") {
+    if (quantileSettings.type === "unit" && quantileSettings.ignoreZeros) {
+      HAVING = `\n-- Ignore zeros in percentile\nHAVING ${numeratorCol} > 0`;
+    }
+  }
+
+  const experimentSQL = `
+SELECT
+  variation,
+  ${
+    type !== "quantile"
+      ? `${
+          type === "proportion" || numerator.column === "$$distinctUsers"
+            ? `-- Number of users who converted`
+            : `-- Total ${type === "ratio" ? "numerator" : "metric"} value`
+        }
+  SUM(m.value) as numerator,
+  ${
+    type === "ratio"
+      ? `-- ${
+          denominator?.column === "$$distinctusers"
+            ? `Number of users who converted`
+            : `Total denominator value`
+        }\n  SUM(d.value)`
+      : `-- Number of users in experiment\n  COUNT(*)`
+  } as denominator,\n  `
+      : ""
+  }${
+    type === "quantile"
+      ? `-- Final result\n  PERCENTILE(${
```

Comment:
```suggestion
      ? `-- Final result\n  APPROX_PERCENTILE(${
```

Just to ward off some questions here, maybe, and be more accurate. Feel free to ignore.

Reply:
Doesn't seem to be any standardization here between engines.  I think I'll leave it as just `PERCENTILE`.
- BigQuery: `APPROX_QUANTILES`
- ClickHouse: `quantile`
- MS SQL: `APPROX_PERCENTILE_CONT`
- Postgres: `PERCENTILE_CONT`
- Vertica: `APPROXIMATE_PERCENTILE`

Reply:
They are all approximations except for postgres, AFAIK.